pkg/standalone-utils/contracts/StableLPOracle.sol

```diff
@@ -0,0 +1,212 @@
+import { LPOracleBase } from "./LPOracleBase.sol";
+
+contract StableLPOracle is LPOracleBase {
+    using FixedPoint for uint256;
+    using SafeCast for *;
+
+    // Thrown when the k parameter did not converge to the positive root.
+    error KDidntConverge();
+
+    int256 private constant _ONE_INT = 1e18;
+    int256 private constant _K_MAX_ERROR = 0.000000000001e16; // 0.000000000001%
+
+    constructor(
+        IVault vault_,
+        IStablePool pool_,
+        AggregatorV3Interface[] memory feeds,
+        uint256 version_
+    ) LPOracleBase(vault_, IBasePool(address(pool_)), feeds, version_) {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    /// @inheritdoc ILPOracleBase
+    function calculateTVL(int256[] memory prices) public view override returns (uint256 tvl) {
+        (, , , uint256[] memory lastBalancesLiveScaled18) = _vault.getPoolTokenInfo(address(pool));
+
+        // The TVL of the stable pool is computed by calculating the balances for the stable pool that would represent
+        // the given price vector. To compute these balances, we need only the amplification parameter of the pool,
+        // the invariant and the price vector.
+
+        uint256 invariant = pool.computeInvariant(lastBalancesLiveScaled18, Rounding.ROUND_DOWN);
+
+        uint256[] memory marketPriceBalancesScaled18 = _computeMarketPriceBalances(invariant, prices);
+
+        tvl = 0;
+        for (uint256 i = 0; i < _totalTokens; i++) {
+            tvl += prices[i].toUint256().mulDown(marketPriceBalancesScaled18[i]);
+        }
+
+        return tvl;
+    }
+
+    /**
+     * @dev Computes a set of balances for a given price vector and a given invariant. The set of balances should meet
+     * two conditions:
+     * 1. The invariant of the computed balances should be equal to the given invariant.
+     * 2. The spot prices of the computed balances should be equal to the given price vector. In other words:
+     *    `spotPrice(i, j) == prices[i] / prices[j]`
+     */
+    function _computeMarketPriceBalances(
+        uint256 invariant,
+        int256[] memory prices
+    ) internal view returns (uint256[] memory balancesForPrices) {
+        // To compute the balances for a given price vector, we need to compute the gradient of the stable invariant.
+        // The stable invariant is:
+        //
+        // a \sum{xi} \prod{xi} +bD \prod{xi} - D^(n+1) = 0
+        //
+        // where `D` is the invariant, `n` is the number of tokens, `xi` is the balance of each token,
+        // `a = A*(n^2n)` and `b = a - n^n`.
+        //
+        // The gradient in terms of xj (the balance of the j-th token) is:
+        //
+        // a \prod{xi} + a \sum{xi} \prod_i!=j{xi} + bD \prod_i!=j{xi}
+        //
+        // We can make this gradient equal to k*pj, where pj is the price of the j-th token and k is a constant.
+        // Then, solving this system of equations for every pj, we will have an array of balances that respect the
```

Comment:
```suggestion
        // Then, solving this system of equations for every pj, we will have an array of balances that reflect the
```
Or "that correspond to the price vector"; we mean that a pool with these balances would have the expected token prices.